service/glacier/internal/customizations/treehash.go

```diff
@@ -0,0 +1,144 @@
+package customizations
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"github.com/awslabs/smithy-go"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+	"io"
+)
+
+const bufsize = 1024 * 1024
```

Comment:
not significant but this could be defined in the `computeHashes` function body as a `const` instead of at the package level. Prevents extra pkg `const`s that aren't clear on their usage.